lib/broadway_dashboard.ex

```diff
@@ -19,13 +19,16 @@ defmodule BroadwayDashboard do
 
   @impl true
   def init(opts) do
-    pipelines = Keyword.get(opts, :pipelines, [])
+    # In case no pipeline is passed in the router, we auto-discover them.
+    pipelines = with [] <- opts[:pipelines] || [], do: :auto_discover
 
     {:ok, %{pipelines: pipelines}, application: :broadway}
   end
 
   @impl true
   def menu_link(%{pipelines: pipelines}, _capabilities) do
+    pipelines = pipelines_or_auto_discover(pipelines)
```

Comment:
I don't think this will work because the current node may not have Broadway. I would do something like this:

    if pipelines == [] do
      {:disabled, ...}
    else
      {:ok, @page_title}
    end